modules/gcs/main.tf

```diff
@@ -108,11 +99,27 @@ resource "google_storage_bucket" "bucket" {
     }
   }
 
-  dynamic "custom_placement_config" {
-    for_each = var.custom_placement_config == null ? [] : [""]
+  dynamic "retention_policy" {
+    for_each = var.retention_policy == null ? [] : [""]
+    content {
+      retention_period = var.retention_policy.retention_period
+      is_locked        = var.retention_policy.is_locked
+    }
+  }
 
+  dynamic "soft_delete_policy" {
```

Comment:
This is the actual added block. 
I resorted all other dynamics - apologies for the diff-spam :)